src/backend/src/utils/change-requests.utils.ts

```diff
@@ -53,7 +55,56 @@ export const sendSlackCRReviewedNotification = async (slackId: string, crId: num
 };
 
 /**
- * Makes sure that a change request has been accepted already (and not deleted)
+ *
+ * @param workPackages The Projects work Packages to be checked
+ * @param wbsElement The wbs element to check for
+ * @param timelineImpact the timeline impact of the proposed solution
+ * @param crId the change request id
+ * @param reviewer the reviewer of the change request
+ *
+ * This function will recursively update the start date of all work packages that depend on the wbs element to be finished
+ */
+export const updateDependencies = async (
+  workPackages: Prisma.Work_PackageGetPayload<typeof workPackageDependencyQueryArgs>[],
+  wbsElement: WBS_Element,
+  timelineImpact: number,
+  crId: number,
+  reviewer: User
+) => {
+  // cycle through all the work packages
+  workPackages.forEach(async (wp) => {
+    // if the work package depends on the wbs element
+    if (wp.dependencies.map((d: WBS_Element) => d.wbsElementId).includes(wbsElement.wbsElementId)) {
+      // update the start date of the work package by the timeline impact
+      const newStartDate = calculateEndDate(wp.startDate, timelineImpact);
+      // create a change to reflect the changing start date
+      const change = {
+        changeRequestId: crId,
+        implementerId: reviewer.userId,
+        detail: buildChangeDetail('Start Date', String(wp.startDate), String(newStartDate))
+      };
+
+      // update the work package
+      await prisma.work_Package.update({
+        where: { workPackageId: wp.workPackageId },
+        data: {
+          startDate: newStartDate,
+          wbsElement: {
+            update: {
+              changes: {
+                create: change
+              }
+            }
+          }
+        }
+      });
+      // recursively update the work packages that depends on the updated work package
+      updateDependencies(workPackages, wp.wbsElement, timelineImpact, crId, reviewer);
```

Comment:
so we honestly shouldn't do recursion because of stack limits. im gonna try to keep this simple but here's how i see it:

```
things to note:
- im skipping a lot of type stuff and so things wont be the exact same
- i dont forget exactly how sets work in typescript but it's something like this
- we are gonna need to test the hell out of this


updateDependencies(initialWorkPackage, crid, reviewer, timelineImpact) {
  const seenWorkPackageIds: Set<number> = [initialWorkPackage.id]; // use a set

  const workPackageUpdateQueue = [initialWorkpackage.dependencies];

  while (workPackageUpdateQueue.length !== 0) {
  	const curr = workPackageUpdateQueue.pop(); // take the first one off

  	// something like this
  	if (seenWorkPackageIds.has(curr.id)) {
  	  continue;
  	}

  	seenWorkPackageIds.add(curr.id);

  	// step 1: update curr's startdate 
  	// step 2: create a change for it

  	// step 3: get all of curr's dependencies
  	// step 4: add all of curr's dependencies to workPackageUpdateQueue
  }
}
```

this is pseudocode for how this function should be. This is basically how non-recursive dfs/bfs works depending on where you pop from and append to. it'll be nice and efficient and stuff

the thing with the set is that if we've updated something already, we skip it. eventually we'll run out of work packages because in the largest case we just see every single work package in the database and stuff. (pop removes things from the list so every time that happens it shrinks until it's empty)